### ISSUE TITLE
Baremetal provisioning

### ISSUE THREAD

**Issue by discostur** [gardener/gardener#614]
Hi,

are there any plans for expanding the baremetal support? As far as i can see you can only provision k8s via openstack in your own datacenter. It would be nice if there is a more flexible way of provisioning clusters in your own datacenter, for example you just provide IP / credentials to your VMs and the provisioning starts. Provisioning the VMs is up to you / your custom virtualization cluster ...

Thanks
Greets
Kilian

**Comment by vlerenc** [comment]
Gardener is actually more about managing fleets of Kubernetes clusters without manual effort. We have support for AWS, Azure, GCP, and Openstack. Alibaba is almost done. Packet has started the work and VMware is on the horizon with the work to start in January '19. However, bare metal in the sense of pet VMs is not fitting the concept of Gardener and will not be supported (as of now). All day-2 operations (e.g. rolling out OS updates) expect cattle VMs and are carried out automatically. All self-healing capabilities expect cattle VMs, too. With pet VMs you don't really benefit from Gardener and a standard imperative tool will do fine (you have to prep the VMs manually, so automation is out of the window). For Gardener to make dealing with clusters effortless, it needs sufficient infrastructure capabilities.

However, we are working together with the folks from KubeVirt to come up with something that may interest you, but this will take more time and is not a quick win. They briefly mentioned it here:
https://kccna18.sched.com/event/Ixrt/intro-kubevirt-bof-daniel-gonzalez-nothnagel-sap-ryan-hallisey-red-hat
Basically, running Gardener clusters via KubeVirt (cattle VMs) on a manually set up Kubernetes, i.e. some kind of lightweight programmable infrastructure for everybody.

**Comment by discostur** [comment]
@vlerenc thanks for the explanation, sound legit to me!
